pkg/kubectl/cmd/delete.go

```
@@ -95,8 +96,37 @@ func RunDelete(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []str
 		return err
 	}
 
+	// By default use a reaper to delete all related resources.
+	if cmdutil.GetFlagBool(cmd, "recursive") {
+		return ReapResult(r, f, out)
+	}
+	return DeleteResult(r, out)
+}
+
+func ReapResult(r *resource.Result, f *cmdutil.Factory, out io.Writer) error {
+	found := 0
+	err := r.Visit(func(info *resource.Info) error {
+		found++
+		reaper, err := f.Reaper(info.Mapping)
+		cmdutil.CheckErr(err)
```

Probably don't want to `fatal` in here.  Can you return the error instead?